### PR TITLE
feat: Log dashboard favorite views DHIS2-9553

### PR DESCRIPTION
**Pull request by jimgrace**
See https://jira.dhis2.org/browse/DHIS2-9553

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-setting/src/main/java/org/hisp/dhis/setting/SettingKey.java [hunk 1]
```diff
@@ -168,6 +168,7 @@
     ANALYTICS_HIDE_WEEKLY_PERIODS( "keyHideWeeklyPeriods", Boolean.FALSE, Boolean.class ),
     ANALYTICS_HIDE_MONTHLY_PERIODS( "keyHideMonthlyPeriods", Boolean.FALSE, Boolean.class ),
     ANALYTICS_HIDE_BIMONTHLY_PERIODS( "keyHideBiMonthlyPeriods", Boolean.FALSE, Boolean.class ),
+    COUNT_DASHBOARD_VIEWS_OF_FAVORITES( "keyCountDashboardViewsOfFavorites", Boolean.FALSE, Boolean.class ),
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-setting/src/main/java/org/hisp/dhis/setting/SettingKey.java:171]: The name could be improved on, as i) we don't refer to favorites rather analytical objects in the source code and ii) this relates to the "data statistics" feature and endpoint, which is about more than simply counting.

Another name could be e.g. `ENABLE_ANALYTICAL_OBJECT_STATISTICS_IN_DASBOARDS`.

**Review by larshelge — CHANGES_REQUESTED** [review]
The name could be improved on, as i) we don't refer to favorites rather analytical objects in the source code and ii) this relates to the "data statistics" feature and endpoint, which is about more than simply counting.

Another name could be e.g. ENABLE_ANALYTICAL_OBJECT_STATISTICS_IN_DASBOARDS.

**Review by stian-sandvold — APPROVED** [review]
Approved, pending resolution of @larshelge 's comment.

**Comment by jimgrace** [comment]
@larshelge Both points well-taken. Just to be clear that we're not enabling the viewing of statistics in dashboards with this setting but gathering them there, I suggest GATHER_ANALYTICAL_OBJECT_STATISTICS_IN_DASHBOARD_VIEWS.

File: dhis-2/dhis-services/dhis-service-setting/src/main/java/org/hisp/dhis/setting/SettingKey.java (hunk 1 above)
Comment (jimgrace) [reply dhis-2/dhis-services/dhis-service-setting/src/main/java/org/hisp/dhis/setting/SettingKey.java]: @larshelge Both points well-taken. Just to be clear that we're not enabling the viewing of statistics in dashboards with this setting but gathering them there, I suggest GATHER_ANALYTICAL_OBJECT_STATISTICS_IN_DASHBOARD_VIEWS.

**Pull request merged**